float32 x
float32 y
float32 z
float32 thta
int32 walking_mode
int32 time_point_
int32 counter
bool stopflag
bool isfirststep
bool islaststep
---
bool cpgack
